services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java

```diff
@@ -23,305 +23,451 @@
+                drawerLayout.closeDrawer(GravityCompat.START);
+            } else if (v.getId() == R.id.btnDrawerLogin) {
+                if (userState == UserState.LOGGED_OUT) {
+                    onSignInButtonClicked();
+                } else {
+                    onSignOutButtonClicked();
+                }
+            }
+        }
+    }
+
+    /**
+     * Class handling actions corresponding to Toolbar-Menu Item Click
+     */
+    private class OnToolbarMenuItemClick implements Toolbar.OnMenuItemClickListener {
+
+        @Override
+        public boolean onMenuItemClick(MenuItem item) {
+            int id = item.getItemId();
+
+            if (id == R.id.action_settings) {
+                Intent intent = new Intent(AbsSyncBaseActivity.this, AppPropertiesActivity.class);
+                intent.putExtra(IntentConsts.INTENT_KEY_APP_NAME, getAppName());
+                startActivityForResult(intent, SETTINGS_ACTIVITY_RESULT_CODE);
+                return true;
+            }
+
+            return false;
+        }
+    }
+
+    /**
+     * Class handling actions corresponding to Drawer Menu Item Click
+     */
+    private class OnDrawerMenuItemClick implements NavigationView.OnNavigationItemSelectedListener {
+
+        @Override
+        public boolean onNavigationItemSelected(MenuItem item) {
+            drawerLayout.closeDrawer(GravityCompat.START);
+
+            if (item.getItemId() == R.id.drawer_resolve_conflict) {
+                Intent i = new Intent(AbsSyncBaseActivity.this, AllConflictsResolutionActivity.class);
+                i.putExtra(IntentConsts.INTENT_KEY_APP_NAME, getAppName());
+                startActivityForResult(i, RESOLVE_CONFLICT_ACTIVITY_RESULT_CODE);
+                return true;
+            } else if (item.getItemId() == R.id.drawer_about_us) {
+                FragmentManager mgr = getSupportFragmentManager();
+                GoToAboutFragment.GotoAboutFragment(mgr, R.id.sync_activity_view);
+                item.setVisible(false);
+                return true;
+            } else if (item.getItemId() == R.id.drawer_settings) {
+                Intent intent = new Intent(AbsSyncBaseActivity.this, AppPropertiesActivity.class);
+                intent.putExtra(IntentConsts.INTENT_KEY_APP_NAME, getAppName());
+                startActivityForResult(intent, SETTINGS_ACTIVITY_RESULT_CODE);
+                return true;
+            } else if (item.getItemId() == R.id.drawer_docs) {
+                Intent browserIntent = new Intent(
+                        Intent.ACTION_VIEW,
+                        Uri.parse(getString(R.string.opendatakit_url))
+                );
+
+                if (browserIntent.resolveActivity(AbsSyncBaseActivity.this.getApplicationContext().getPackageManager()) != null) {
+                    startActivity(browserIntent);
+                } else {
+                    Intent i = new Intent(AbsSyncBaseActivity.this, DocumentationWebViewActivity.class);
```

Comment:
If making changes in legacy code, I think we get a chance to fix lot of things like naming conventions, thread optimization, code paradigms, etc. Not necessarily now but it is something that we should look into in future.